### DISCUSSION TITLE
Software Heritage archival of RPM source code

### DISCUSSION THREAD

**Discussion by benoit-chauvet** [rpm-software-management/rpm#2694] [General]
Hello,

Software Heritage is the universal archive of source code, and a non profit organization.
Our mission is to collect, preserve and share source code from every possible origins, 
including forges and package managers.

Software Heritage will soon be ready to support the ingestion of RPM source code
modules. The RPM loader has been deployed in our staging infrastructure [1].
We have ingested origins that were listed from Fedora and CentOS [2].

We now would like to have a round of public review on the actual data output
of the ingestion process, so we can either validate or ask for some more
updates, and then eventually deploy it in production.

Still, for now, please have a look at some of those origins loaded [2]
and let us know what you think!

[1] https://gitlab.softwareheritage.org/swh/meta/-/issues/5042
[2] https://webapp.staging.swh.network/browse/search/?q=&with_visit=true&with_content=true&visit_type=rpm

Best regards,

Benoit Chauvet
Software Engineering Manager
Software Heritage

**Comment by pmatilai** [comment]
Hi, and thanks for your interest!

I think there's some terminology confusion here: the rpm source code is in this GH repository. We'd certainly want it archived too.

The material in the second link is known as "RPM spec files", which are essentially build recipes for other software packages. It does seem like a good idea to archive that kind of material as well, but the rpm upstream doesn't "own" that material in any sense of the word. So it might be better to ask for input in the communities whose spec files are being collected: Fedora and CentOS.
